### PR TITLE
GLT-3102: allow changing sequencing container model

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultContainerService.java [hunk 1]
```diff
@@ -122,6 +129,32 @@ private void validateChange(SequencerPartitionContainer container, SequencerPart
...
+        Stream<InstrumentModel> requiredInstrumentModels = beforeChange.getRunPositions().stream()
+            .map(RunPosition::getRun)
+            .map(Run::getSequencer)
+            .map(Instrument::getInstrumentModel);
+        if (requiredInstrumentModels.anyMatch(required -> after.getInstrumentModels().stream()
+            .map(InstrumentModel::getId)
+            .noneMatch(id -> id == required.getId()))) {
+          errors.add(new ValidationError("model.id",
+              String.format("Can only change to a model compatible with the linked runs' instrument models (%s)",
+                  LimsUtils.joinWithConjunction(requiredInstrumentModels.map(InstrumentModel::getAlias).collect(Collectors.toSet()),
```
Comment (apmasell) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultContainerService.java]: Streams aren't supposed to be reused. Maybe create a list or set of instrument models?

Comment (djcooke) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultContainerService.java]: fixed

File: miso-web/src/main/webapp/scripts/form_container.js [hunk 2]
```diff
@@ -40,6 +36,10 @@ FormTarget.container = (function() {
           getDisplayValue: function(container) {
             return container.model.alias;
           }
+        }, {
+          title: 'Change Model',
+          type: 'special',
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/form_container.js]: (Just leaving a note here that we spoke earlier about converting the container model field to a dropdown instead.)

Comment (djcooke) [reply miso-web/src/main/webapp/scripts/form_container.js]: Changed. Dropdown will sometimes contain a mix of models from multiple instruments when the container isn't connected to any runs, but it should never be too long of a list

**Pull request merged**